ic-auction/src/state.rs

```diff
@@ -0,0 +1,206 @@
+use std::collections::HashMap;
+
+use ic_canister::ic_kit::ic;
+use ic_cdk::export::candid::{CandidType, Deserialize, Principal};
+use ic_helpers::metrics::Interval;
+use ic_storage::IcStorage;
+
+use crate::{
+    error::{AuctionError, Result},
+    AuctionInfo, BiddingInfo, Cycles, Timestamp,
+};
+
+// Minimum bidding amount is required, for every update call costs cycles, and we want bidding
+// to add cycles rather then to decrease them. 1M is chosen as one ingress call costs 590K cycles.
+pub const MIN_BIDDING_AMOUNT: Cycles = 1_000_000;
+
+//--------------------------------------------------------------------------------------------------
+// Bidding state
+//--------------------------------------------------------------------------------------------------
```

Comment:
Love the comment style, but cap it at around 80 chars
```suggestion
//------------------------------------------------------------------------------
// Bidding state
//------------------------------------------------------------------------------
```

Reply:
Strange, I thought copilot was smart enough setting it to 80 chars

Reply:
COPILOT!!!!! D: